### ISSUE TITLE
"sentry_project" does not exist

### ISSUE THREAD

**Issue by ebisbe** [getsentry/sentry#1578]
Hello,

I got this error on the title doing the migrations. I've searched a bit and all the issues are closed and seem resolved. I'm trying to install Sentry 7.6
The sentry.conf.py is the same that generates the sentry with all the users and passwords updated ( I don't think is really needed ).
```
# SENTRY_CONF=/etc/sentry.conf.py sentry upgrade
Syncing...
Creating tables ...
Creating table django_admin_log
Creating table auth_permission
Creating table auth_group_permissions
Creating table auth_group
Creating table django_content_type
Creating table django_session
Creating table django_site
Creating table south_migrationhistory
Traceback (most recent call last):
  File "/usr/local/bin/sentry", line 9, in <module>
    load_entry_point('sentry==7.6.0', 'console_scripts', 'sentry')()
  File "/usr/local/lib/python2.7/dist-packages/sentry/utils/runner.py", line 428, in main
    initializer=initialize_app,
  File "/usr/local/lib/python2.7/dist-packages/logan/runner.py", line 193, in run_app
    management.execute_from_command_line([runner_name, command] + command_args)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 399, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 392, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 242, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 285, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python2.7/dist-packages/sentry/management/commands/upgrade.py", line 34, in handle
    verbosity=options['verbosity'],
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 159, in call_command
    return klass.execute(*args, **defaults)
  File "/usr/local/lib/python2.7/dist-packages/raven/contrib/django/management/__init__.py", line 46, in new_execute
    'argv': sys.argv
  File "/usr/local/lib/python2.7/dist-packages/raven/base.py", line 679, in captureException
    'raven.events.Exception', exc_info=exc_info, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sentry/utils/raven.py", line 39, in capture
    return super(SentryInternalClient, self).capture(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/raven/contrib/django/client.py", line 157, in capture
    result = super(DjangoClient, self).capture(event_type, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/raven/base.py", line 526, in capture
    self.send(**data)
  File "/usr/local/lib/python2.7/dist-packages/sentry/utils/raven.py", line 50, in send
    project = Project.objects.get_from_cache(id=settings.SENTRY_PROJECT)
  File "/usr/local/lib/python2.7/dist-packages/sentry/db/models/manager.py", line 209, in get_from_cache
    result = self.get(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/manager.py", line 151, in get
    return self.get_queryset().get(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 304, in get
    num = len(clone)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 77, in __len__
    self._fetch_all()
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 857, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 220, in iterator
    for row in compiler.results_iter():
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/sql/compiler.py", line 713, in results_iter
    for rows in self.execute_sql(MULTI):
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/sql/compiler.py", line 786, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python2.7/dist-packages/django/db/utils.py", line 99, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python2.7/dist-packages/sentry/db/postgres/decorators.py", line 75, in inner
    raise_the_exception(self.db, e)
  File "/usr/local/lib/python2.7/dist-packages/sentry/db/postgres/decorators.py", line 73, in inner
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sentry/db/postgres/decorators.py", line 21, in inner
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sentry/db/postgres/decorators.py", line 95, in inner
    six.reraise(exc_info[0], exc_info[0](msg), exc_info[2])
  File "/usr/local/lib/python2.7/dist-packages/sentry/db/postgres/decorators.py", line 88, in inner
    return func(self, sql, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sentry/db/postgres/base.py", line 41, in execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: relation "sentry_project" does not exist
LINE 1: ...oject"."status", "sentry_project"."platform" FROM "sentry_pr...
                                                             ^

SQL: SELECT "sentry_project"."id", "sentry_project"."slug", "sentry_project"."name", "sentry_project"."organization_id", "sentry_project"."team_id", "sentry_project"."public", "sentry_project"."date_added", "sentry_project"."status", "sentry_project"."platform" FROM "sentry_project" WHERE "sentry_project"."id" = %s
```

**Comment by sergey-stukov** [comment]
Have the same issue sentry==7.6.0

**Comment by dcramer** [comment]
I have a feeling whats happening here is something else is erroring and this is covering up the exception.

Is this a brand new install?

**Comment by ebisbe** [comment]
Yes, a new installation. Now is working but with sqllite instead of postgresql. But now I haven't achieved to start it.

**Comment by luxifer** [comment]
same here...

**Comment by luxifer** [comment]
```
(sentry)xotelia@sd-66810:~$ sentry upgrade -v 3
Syncing...
Running pre-sync handlers for application admin
Running pre-sync handlers for application auth
Running pre-sync handlers for application contenttypes
Running pre-sync handlers for application messages
Running pre-sync handlers for application sessions
Running pre-sync handlers for application sites
Running pre-sync handlers for application staticfiles
Running pre-sync handlers for application captcha
Running pre-sync handlers for application crispy_forms
Running pre-sync handlers for application raven_compat
Running pre-sync handlers for application rest_framework
Running pre-sync handlers for application sentry_interface_types
Running pre-sync handlers for application sentry_mail
Running pre-sync handlers for application sentry_urls
Running pre-sync handlers for application sentry_useragents
Running pre-sync handlers for application sentry_webhooks
Running pre-sync handlers for application south
Running pre-sync handlers for application sudo
Creating tables ...
Running post-sync handlers for application admin
Traceback (most recent call last):
  File "/home/xotelia/sentry/bin/sentry", line 11, in <module>
    sys.exit(main())
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/sentry/utils/runner.py", line 428, in main
    initializer=initialize_app,
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/logan/runner.py", line 193, in run_app
    management.execute_from_command_line([runner_name, command] + command_args)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 399, in execute_from_command_line
    utility.execute()
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 392, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/core/management/base.py", line 242, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/core/management/base.py", line 285, in execute
    output = self.handle(*args, **options)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/sentry/management/commands/upgrade.py", line 34, in handle
    verbosity=options['verbosity'],
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 159, in call_command
    return klass.execute(*args, **defaults)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/raven/contrib/django/management/__init__.py", line 46, in new_execute
    'argv': sys.argv
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/raven/base.py", line 679, in captureException
    'raven.events.Exception', exc_info=exc_info, **kwargs)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/sentry/utils/raven.py", line 39, in capture
    return super(SentryInternalClient, self).capture(*args, **kwargs)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/raven/contrib/django/client.py", line 157, in capture
    result = super(DjangoClient, self).capture(event_type, **kwargs)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/raven/base.py", line 526, in capture
    self.send(**data)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/sentry/utils/raven.py", line 50, in send
    project = Project.objects.get_from_cache(id=settings.SENTRY_PROJECT)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/sentry/db/models/manager.py", line 209, in get_from_cache
    result = self.get(**kwargs)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/db/models/manager.py", line 151, in get
    return self.get_queryset().get(*args, **kwargs)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/db/models/query.py", line 304, in get
    num = len(clone)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/db/models/query.py", line 77, in __len__
    self._fetch_all()
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/db/models/query.py", line 857, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/db/models/query.py", line 220, in iterator
    for row in compiler.results_iter():
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 713, in results_iter
    for rows in self.execute_sql(MULTI):
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 786, in execute_sql
    cursor.execute(sql, params)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/db/utils.py", line 99, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/sentry/db/postgres/decorators.py", line 75, in inner
    raise_the_exception(self.db, e)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/sentry/db/postgres/decorators.py", line 73, in inner
    return func(self, *args, **kwargs)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/sentry/db/postgres/decorators.py", line 21, in inner
    return func(self, *args, **kwargs)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/sentry/db/postgres/decorators.py", line 95, in inner
    six.reraise(exc_info[0], exc_info[0](msg), exc_info[2])
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/sentry/db/postgres/decorators.py", line 88, in inner
    return func(self, sql, *args, **kwargs)
  File "/home/xotelia/sentry/local/lib/python2.7/site-packages/sentry/db/postgres/base.py", line 41, in execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: relation "sentry_project" does not exist
LINE 1: ...oject"."status", "sentry_project"."platform" FROM "sentry_pr...
                                                             ^

SQL: SELECT "sentry_project"."id", "sentry_project"."slug", "sentry_project"."name", "sentry_project"."organization_id", "sentry_project"."team_id", "sentry_project"."public", "sentry_project"."date_added", "sentry_project"."status", "sentry_project"."platform" FROM "sentry_project" WHERE "sentry_project"."id" = %s
```

**Comment by luxifer** [comment]
I just did `sentry migrate` and then `sentry upgrade` and it works fine. I think the `upgrade` task runs also handlers, which some of them requires specific tables.

**Comment by osama-haggag** [comment]
Hi, I was having the same issue. But my redis-server wasn't running. Just running the `redis-server` made the `sentry upgrade` command work straight off.

**Comment by sergey-stukov** [comment]
I have success with install by downgrading to 7.5.4, but seems key to solution is to run redis-server as said @osama-haggag

**Comment by dcramer** [comment]
Downgrading generally doesn't fix things in Sentry (and usually breaks more than it fixes). You definitely are required to run Redis server, and should probably skim the getting started guide to make sure nothing else is missing.

Closing unless there's another issue here.

**Comment by luxifer** [comment]
Is there a place where I can update the doc?

**Comment by dcramer** [comment]
If anything we should raise a runtime error if this is misconfigured or something.

Was it purely that it couldn't connect or that it wasn't setup correctly?

**Comment by deric** [comment]
Same issue with Sentry 7.7.0 and clean of PostgreSQL database. I have redis server running and using either `sentry upgrade` nor `sentry upgrade` does not help.

Any ideas? Is there a command for creating initial schema?

```
SENTRY_CONF=/etc/sentry.conf.py sentry upgrade
Syncing...
Creating tables ...
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/raven/base.py", line 211, in handle_exception
    self.captureException(exc_info=exc_info)
  File "/usr/local/lib/python2.7/dist-packages/raven/base.py", line 676, in captureException
    'raven.events.Exception', exc_info=exc_info, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sentry/utils/raven.py", line 39, in capture
    return super(SentryInternalClient, self).capture(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/raven/contrib/django/client.py", line 157, in capture
    result = super(DjangoClient, self).capture(event_type, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/raven/base.py", line 521, in capture
    self.send(**data)
  File "/usr/local/lib/python2.7/dist-packages/sentry/utils/raven.py", line 50, in send
    project = Project.objects.get_from_cache(id=settings.SENTRY_PROJECT)
  File "/usr/local/lib/python2.7/dist-packages/sentry/db/models/manager.py", line 209, in get_from_cache
    result = self.get(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/manager.py", line 151, in get
    return self.get_queryset().get(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 304, in get
    num = len(clone)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 77, in __len__
    self._fetch_all()
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 857, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 220, in iterator
    for row in compiler.results_iter():
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/sql/compiler.py", line 713, in results_iter
    for rows in self.execute_sql(MULTI):
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/sql/compiler.py", line 786, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python2.7/dist-packages/django/db/utils.py", line 99, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: relation "sentry_project" does not exist
LINE 1: ...oject"."status", "sentry_project"."platform" FROM "sentry_pr...
                                                             ^
```

**Comment by dcramer** [comment]
'sentry upgrade' creates the initial schema, but somethings erroring and its not handling it well. Try installing from master.

tl;dr you have something misconfigured and its erroring, but in this version of sentry we are shitty at handling that case

**Comment by deric** [comment]
I've copied configuration line from previous version:
```
'ENGINE': 'django.db.backends.postgresql_psycopg2',
```
instead of:
```
'ENGINE': 'sentry.db.postgres',
```
still, the error persists. Using
```
SENTRY_CONF=/etc/sentry.conf.py sentry syncdb --all
```
I was able to create at least an account. However, migrations still fails:
```
  File "/usr/local/lib/python2.7/dist-packages/sentry/../south/migration/base.py", line 312, in migration
    raise exceptions.UnknownMigration(self, sys.exc_info())
south.exceptions.UnknownMigration: Migration 'admin:0001_initial' probably doesn't exist.
...
  File "/usr/local/lib/python2.7/dist-packages/django/apps/config.py", line 4, in <module>
    from django.core.exceptions import AppRegistryNotReady, ImproperlyConfigured
ImportError: cannot import name AppRegistryNotReady

```

**Comment by dcramer** [comment]
I'm not sure, but you might have a wrong version of Django:

``pip freeze``

Make sure you're doing the install exactly as the instructions dictate and not installing any odd versions of Django/etc

**Comment by dcramer** [comment]
Also if you install master I believe we'll handle the error you're seeing safely.

**Comment by deric** [comment]
I've created a clean virtualenv and installed a version from master - `7.8.0-dev`. Now it finally works, thanks! Using `SENTRY_CONF=/etc/sentry.conf.py sentry upgrade` was enough.

**Comment by timeseer** [comment]
I have same issue with sentry==8.0.1 and PostgreSQL 9.2.14
These are my db configuration and error message when I input `sentry upgrade` to shell.

```
DATABASES = {
    'default': {
        'ENGINE': 'sentry.db.postgres',
        'NAME': 'sentry',
        'USER': 'postgres',
        'PASSWORD': 'MYDBPASSWORD',
        'HOST': '',
        'PORT': '',
    }
}
```

```
[ERROR] Unable to fetch internal project
Traceback (most recent call last):
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/utils/raven.py", line 59, in send
    project = Project.objects.get_from_cache(id=settings.SENTRY_PROJECT)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/db/models/manager.py", line 259, in get_from_cache
    result = self.get(**kwargs)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/django/db/models/manager.py", line 151, in get
    return self.get_queryset().get(*args, **kwargs)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/django/db/models/query.py", line 304, in get
    num = len(clone)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/django/db/models/query.py", line 77, in __len__
    self._fetch_all()
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/django/db/models/query.py", line 857, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/django/db/models/query.py", line 220, in iterator
    for row in compiler.results_iter():
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 713, in results_iter
    for rows in self.execute_sql(MULTI):
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 786, in execute_sql
    cursor.execute(sql, params)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/django/db/utils.py", line 99, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/db/postgres/decorators.py", line 75, in inner
    raise_the_exception(self.db, e)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/db/postgres/decorators.py", line 73, in inner
    return func(self, *args, **kwargs)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/db/postgres/decorators.py", line 21, in inner
    return func(self, *args, **kwargs)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/db/postgres/decorators.py", line 95, in inner
    six.reraise(exc_info[0], exc_info[0](msg), exc_info[2])
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/db/postgres/decorators.py", line 88, in inner
    return func(self, sql, *args, **kwargs)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/db/postgres/base.py", line 44, in execute
    return self.cursor.execute(sql, params)
ProgrammingError: relation "sentry_project" does not exist
LINE 1: ...ct"."status", "sentry_project"."first_event" FROM "sentry_pr...
                                                             ^

SQL: SELECT "sentry_project"."id", "sentry_project"."slug", "sentry_project"."name", "sentry_project"."organization_id", "sentry_project"."team_id", "sentry_project"."public", "sentry_project"."date_added", "sentry_project"."status", "sentry_project"."first_event" FROM "sentry_project" WHERE "sentry_project"."id" = %s
Traceback (most recent call last):
  File "/home/user/.virtualenvs/sentry/bin/sentry", line 11, in <module>
    sys.exit(main())
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/runner/__init__.py", line 136, in main
    cli(prog_name=get_prog(), obj={}, max_content_width=100)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/runner/decorators.py", line 25, in inner
    configure()
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/runner/__init__.py", line 99, in configure
    configure(ctx, py, yaml, skip_backend_validation)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/runner/settings.py", line 317, in configure
    }, skip_backend_validation=skip_backend_validation)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/runner/initializer.py", line 155, in initialize_app
    validate_options(settings)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/runner/initializer.py", line 186, in validate_options
    default_manager.validate(settings.SENTRY_OPTIONS)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/options/manager.py", line 217, in validate
    self.validate_option(k, v)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/options/manager.py", line 220, in validate_option
    opt = self.lookup_key(key)
  File "/home/user/.virtualenvs/sentry/lib/python2.7/site-packages/sentry/options/manager.py", line 103, in lookup_key
    raise UnknownOption(key)
sentry.options.manager.UnknownOption: 's\x00y\x00s\x00t\x00e\x00m\x00.\x00s\x00e'
```


I created 'sentry' database via command `sudo -u postgres createdb -E utf-8 sentry`.
It seems that `upgrade` function in `sentry/runner/commands/upgrade.py` does not called.

**Comment by mattrobenolt** [comment]
What's in your `SENTRY_OPTIONS` in `sentry.conf.py` ? It looks like you have something invalid in there.

**Comment by timeseer** [comment]
There are no `SENTRY_OPTIONS` variable in `sentry.conf.py`. I get this file via command `sentry init`, and I didn't delete that variable.

**Comment by mattrobenolt** [comment]
Did you add anything to `config.yml`?

The problem here is: `sentry.options.manager.UnknownOption: 's\x00y\x00s\x00t\x00e\x00m\x00.\x00s\x00e'`, so I'm trying to figure out where this is coming from.

It's validation over your config files.

**Comment by timeseer** [comment]
Nope. I didn't edit `config.yml` at all. This is my `config.yml`.

```
# While a lot of configuration in Sentry can be changed via the UI, for all
# new-style config (as of 8.0) you can also declare values here in this file
# to enforce defaults or to ensure they cannot be changed via the UI. For more
# information see the Sentry documentation.

# If this file ever becomes compromised, it's important to regenerate your SECRET_KEY
# Changing this value will result in all current sessions being invalidated
system.secret-key: '######################################'
```
I replace my secret-key with `#`.

**Comment by mattrobenolt** [comment]
Ok, so I'm fairly confident there's an issue with reading that `system.secret-key` value from disk in the wrong encoding somehow. I haven't seen this issue before, so not sure how that's happening, but I can investigate.

Can you try removing that line from `config.yml` and instead putting it in `sentry.conf.py`?  Just shove at the bottom:

```
SECRET_KEY = '######'
```

This can just workaround this bug in the meantime while I investigate how this could happen in the first place.

**Comment by timeseer** [comment]
Ok :D It seem to be working now. Thanks you!

**Comment by mattrobenolt** [comment]
I'm going to open a separate issue for this encoding issue.

**Comment by NoahSlv** [comment]
I have the exact same issue as timseer described above.

Have tried moving SECRET_KEY to sentry.conf.py file, but that has no effect. 

It appears as if sentry is failing to create the initial tables.

Full system output is below:

`SENTRY_CONF=/etc/sentry sentry upgrade -v 3
[ERROR] Unable to fetch internal project
Traceback (most recent call last):
  File "/root/sentry/src/sentry/utils/raven.py", line 59, in send
    project = Project.objects.get_from_cache(id=settings.SENTRY_PROJECT)
  File "/root/sentry/src/sentry/db/models/manager.py", line 259, in get_from_cache
    result = self.get(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/manager.py", line 151, in get
    return self.get_queryset().get(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 304, in get
    num = len(clone)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 77, in __len__
    self._fetch_all()
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 857, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 220, in iterator
    for row in compiler.results_iter():
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/sql/compiler.py", line 713, in results_iter
    for rows in self.execute_sql(MULTI):
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/sql/compiler.py", line 786, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python2.7/dist-packages/django/db/utils.py", line 99, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/root/sentry/src/sentry/db/postgres/decorators.py", line 75, in inner
    raise_the_exception(self.db, e)
  File "/root/sentry/src/sentry/db/postgres/decorators.py", line 73, in inner
    return func(self, *args, **kwargs)
  File "/root/sentry/src/sentry/db/postgres/decorators.py", line 21, in inner
    return func(self, *args, **kwargs)
  File "/root/sentry/src/sentry/db/postgres/decorators.py", line 95, in inner
    six.reraise(exc_info[0], exc_info[0](msg), exc_info[2])
  File "/root/sentry/src/sentry/db/postgres/decorators.py", line 88, in inner
    return func(self, sql, *args, **kwargs)
  File "/root/sentry/src/sentry/db/postgres/base.py", line 39, in execute
    return self.cursor.execute(sql, params)
ProgrammingError: relation "sentry_project" does not exist
LINE 1: ...ct"."status", "sentry_project"."first_event" FROM "sentry_pr...
                                                             ^

SQL: SELECT "sentry_project"."id", "sentry_project"."slug", "sentry_project"."name", "sentry_project"."callsign", "sentry_project"."forced_color", "sentry_project"."organization_id", "sentry_project"."team_id", "sentry_project"."public", "sentry_project"."date_added", "sentry_project"."status", "sentry_project"."first_event" FROM "sentry_project" WHERE "sentry_project"."id" = %s 
Traceback (most recent call last):
  File "/usr/local/bin/sentry", line 9, in <module>
    load_entry_point('sentry', 'console_scripts', 'sentry')()
  File "/root/sentry/src/sentry/runner/__init__.py", line 148, in main
    cli(prog_name=get_prog(), obj={}, max_content_width=100)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/root/sentry/src/sentry/runner/decorators.py", line 25, in inner
    configure()
  File "/root/sentry/src/sentry/runner/__init__.py", line 111, in configure
    configure(ctx, py, yaml, skip_backend_validation)
  File "/root/sentry/src/sentry/runner/settings.py", line 325, in configure
    }, skip_backend_validation=skip_backend_validation)
  File "/root/sentry/src/sentry/runner/initializer.py", line 162, in initialize_app
    validate_backends()
  File "/root/sentry/src/sentry/runner/initializer.py", line 185, in validate_backends
    backend.validate()
  File "/root/sentry/src/sentry/buffer/redis.py", line 37, in validate
    raise InvalidConfiguration(unicode(e))
sentry.exceptions.InvalidConfiguration: MISCONF Redis is configured to save RDB snapshots, but is currently not able to persist on disk. Commands that may modify the data set are disabled. Please check Redis logs for details about the error.`

**Comment by mattrobenolt** [comment]
@NoahSlv you have an entirely different issue. It's having a problem talking to Redis. 

> MISCONF Redis is configured to save RDB snapshots, but is currently not able to persist on disk. Commands that may modify the data set are disabled. Please check Redis logs for details about the error.

This is your issue.
